sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/ReduceNumShufflePartitions.scala

```diff
@@ -180,25 +180,45 @@ case class ReduceNumShufflePartitions(conf: SQLConf) extends Rule[SparkPlan] {
 
 case class CoalescedShuffleReaderExec(
     child: QueryStageExec,
-    partitionStartIndices: Array[Int]) extends UnaryExecNode {
+    partitionStartIndices: Array[Int],
+    var isLocal: Boolean = false) extends UnaryExecNode {
 
   override def output: Seq[Attribute] = child.output
 
   override def doCanonicalize(): SparkPlan = child.canonicalized
 
   override def outputPartitioning: Partitioning = {
```

Comment:
Don't we need to override requiredChildDistribution, if isLocal is true?

I saw you check if there are additional shuffle exchange added by EnsureRequirements, to decide if local shuffle reader works or not. If don't change requiredChildDistribution, will EnsureRequirements bring additional shuffle exchange?

Maybe I miss anything here?

Reply:
@viirya 
Maybe not override `requiredChildDistribution`. Because the `requiredChildDistribution` of `CoalescedShuffleReaderExec` is `UnspecificedDistribution` whether the `isLocal ` is `true `or `false`, the `EnsureRequirements ` will not introduce the additional shuffle exchange.

Reply:
Ur, don't you rely on see if EnsureRequirements introduces additional shuffle exchange, to decide doing local shuffle reader or not?

Reply:
Yes I need.